ZenPacks/zenoss/LinuxMonitor/parsers/linux/diskstats.py

```diff
@@ -1,140 +1,196 @@
+         253       6 dm-6 196 0 1568 354 1 0 8 0 0 203 354
+         253       7 dm-7 7 0 56 19 4 0 32 1 0 20 20
+         253       8 dm-8 193 0 1544 260 1 0 8 0 0 260 260
+         253       9 dm-9 822 0 1536 2108 5 0 7 1 0 289 2109
+         253      10 dm-10 6 0 48 19 6 0 48 1 0 20 20
+
 """
-import re
-from collections import namedtuple
-from Products.ZenRRD.CommandParser import CommandParser
+
 
 import logging
-log = logging.getLogger('zen.command.parsers.diskstats')
-
-# datafields in /proc/diskstats, per http://www.kernel.org/doc/Documentation/iostats.txt
-DiskStatsData = namedtuple("DiskStatsData", 
-    "total_reads  num_reads_merged  total_sectors_read    total_ms_reading "
-    "total_writes num_writes_merged total_sectors_written total_ms_writing "
-    "num_io_in_progess  total_ms_io  weighted_ms_per_io")
-
-OUTPUT_DATA_KEYS = ["ssIORawReceived", "ssIORawSent"]
-
-_reCache = {}
-def _getReMatcher(regex):
-    if regex not in _reCache:
-        try:
-            _reCache[regex] = re.compile(regex).match
-        except Exception:
-            _reCache[regex] = None
-    return _reCache[regex]        
-
-def _parseOutput(output, hdFilterFn=None):
-    """
-    Given "/proc/diskstats" output, filter out logical devices and find the
-    sum read sectors and write sectors of the physical devices. Return a
-    mapping of "ssIORawReceived" to that line's reads sectors and 
-    "ssIORawSent" to that line's writes sectors.
-    """
-    # if no filtering function given, accept anything
-    if hdFilterFn is None:
-        hdFilterFn = lambda s: True
-
-    totalReadSectors = 0
-    totalWrittenSectors = 0
-    
-    for line in output.strip().splitlines():
-
-        words = line.split()
-
-        # some logical devices have length less than 14
-        # other logical devices start with dm-
-        if len(words) == 14:
-            diskid = words[2]
-            if hdFilterFn(diskid) and not diskid.startswith("dm-"):
-                diskstatsdata = DiskStatsData(*words[3:])
-                log.debug("extracted data for device '%s': %s", diskid, diskstatsdata)
-                totalReadSectors    += int(diskstatsdata.total_sectors_read)
-                totalWrittenSectors += int(diskstatsdata.total_sectors_written)
-    
-    return dict(zip(OUTPUT_DATA_KEYS, (totalReadSectors, totalWrittenSectors)))
-
-class diskstats(CommandParser):
-
-    HARD_DISK_MATCH_PROPERTY = 'zHardDiskMapMatch'
-
-    def dataForParser(self, context, datapoint):
-        ret = super(diskstats, self).dataForParser(context, datapoint)
-
-        # save global zHardDiskMapMatch filtering regex property, 
-        # to be copied to each datapoint's .data attribute
-        ret['hdFilterRegex'] = getattr(context, self.HARD_DISK_MATCH_PROPERTY, None)
-
-        return ret
+import re
+
+from Products.ZenRRD.ComponentCommandParser import ComponentCommandParser
+
+
+PARTITION = r'(?P<component>[a-z]+\d+\s)'
```

Comment:
Having the `\d+` as part of the match is causing my "xvdb" PV to be missed. I've added /dev/xvdb as a PV directly without partitioning it. This is a valid, and fairly common usage of LVM.